src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -72,8 +74,25 @@ public RuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
     /// <returns>The RuntimeConfig instance.</returns>
     /// <remark>Dont use this method if environment variable references need to be retained.</remark>
     /// <exception cref="DataApiBuilderException">Thrown when the loader is unable to load an instance of the config from its known location.</exception>
-    public RuntimeConfig GetConfig()
+    public RuntimeConfig GetConfig(bool isNewConfigDetected = false, bool isNewConfigValidated = true)
     {
+        //Only used in hot reload to validate the configuration file
+        if (isNewConfigDetected && !isNewConfigValidated)
+        {
+            IFileSystem fileSystem = new FileSystem();
+            ILoggerFactory loggerFactory = new LoggerFactory();
+            ILogger<RuntimeConfigValidator> logger = loggerFactory.CreateLogger<RuntimeConfigValidator>();
+            RuntimeConfigValidator runtimeConfigValidator = new(this, fileSystem, logger, true);
```

Comment:
Not sure if this is fine, since there is no logger inside the provider

Reply:
For now, this is the only straightforward way of achieving validation during a hot reload.

My original response was the following, however we need to re-evaluate how we configure services in startup.cs such that we don't need to have runtimeconfigprovider resolved. That will need to be tracked via a separate task.
--------------------Obsolete-----------------------
There looks to be a better way of accomplishing instantiation of the `RuntimeConfigValidator`.

In Startup.cs:
```csharp
            IFileSystem fileSystem = new FileSystem();
            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, hotReloadEventHandler, configFileName, connectionString);
            RuntimeConfigProvider configProvider = new(configLoader);
            RuntimeConfigValidator configValidator = new(
                runtimeConfigProvider: configProvider, 
                fileSystem: fileSystem, 
                logger: , // this is an issue, we can't resolve an ilogger in Startup::ConfigureServices
                isValidateOnly: true);
            services.AddSingleton(fileSystem);
            services.AddSingleton(configProvider);
            services.AddSingleton(configLoader);
            services.AddSingleton<configValidator>(); // instead of "services.AddSingleton<RuntimeConfigValidator>();"

Reply:
I was not able to find the code you put from Startup.cs and that way of instantiating `RuntimeConfigValidator` did not work for me.

Reply:
Right, that's what I meant by "obsolete." The code sample i provided isn't possible, but in an ideal world, it would be. To be doable, other refactors are necessary. That's the background behind my comment of "For now, this (your current code change) is the only straightforward way of achieving validation during a hot reload."